src/main/java/org/cyclonedx/gradle/utils/EnvironmentUtils.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of CycloneDX Gradle Plugin.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.gradle.utils;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Utility class for working with environment variables in common CI environments.
+ */
+public class EnvironmentUtils {
+
+    /**
+     * Pattern to match environment variables in a string.
+     */
+    private static final Pattern VARIABLE_PATTERN = Pattern.compile("\\$\\{([^}]+)}");
+    /**
+     * Jenkins build URL environment variable.
+     */
+    private static final String JENKINS_BUILD_URL = "BUILD_URL";
+    /**
+     * Travis build URL environment variable.
+     */
+    private static final String TRAVIS_BUILD_WEB_URL = "TRAVIS_BUILD_WEB_URL";
+    /**
+     * CircleCI build URL environment variable.
+     */
+    private static final String CIRCLE_BUILD_URL = "CIRCLE_BUILD_URL";
+    /**
+     * Drone build link environment variable.
+     */
+    private static final String DRONE_BUILD_LINK = "DRONE_BUILD_LINK";
+
+    /**
+     * Get the URI of the current build from the environment variables set on common build systems like GitHub Actions, GitLab CI, etc.
+     * @return the URI of the current build or null if it cannot be determined
+     */
+    public static String getBuildURI() {
+        return Optional.ofNullable(fromGithubActions()).orElseGet(() -> Optional.ofNullable(fromGitlabCI())
+                .orElseGet(() -> Optional.ofNullable(fromEnvironment(JENKINS_BUILD_URL))
+                        .orElseGet(() -> Optional.ofNullable(fromEnvironment(CIRCLE_BUILD_URL))
+                                .orElseGet(() -> Optional.ofNullable(fromEnvironment(TRAVIS_BUILD_WEB_URL))
+                                        .orElseGet(() -> Optional.ofNullable(fromEnvironment(DRONE_BUILD_LINK))
+                                                .orElse(null))))));
```

Comment:
For this and other methods in this class. Method must have `@Nullable` annotation if it can return `null`.

Reply:
done